crawler/db/dart.py

```diff
@@ -146,20 +149,37 @@ def add_dart_plate_if_doesnt_exist(cursor: pyodbc.Cursor, plate_barcode: str, bi
     return state
 
 
-def add_dart_well_properties_if_positive(cursor: pyodbc.Cursor, sample: SampleDoc, plate_barcode: str) -> None:
-    """Adds well properties to DART for the specified sample if that sample is positive.
+def add_dart_well_properties(cursor: pyodbc.Cursor, sample: SampleDoc, plate_barcode: str) -> None:
+    """Adds well properties to DART for the specified sample
+        regardless of if it is important
+        as fields may have been updated to not being important
+        and these need to be updated in Dart
 
     Arguments:
         cursor {pyodbc.Cursor} -- The cursor with which to execute queries.
         sample {Sample} -- The sample for which to add well properties.
         plate_barcode {str} -- The barcode of the plate to which this sample belongs.
     """
-    if sample[FIELD_RESULT] == POSITIVE_RESULT_VALUE:
-        well_index = get_dart_well_index(str(sample.get(FIELD_COORDINATE)))
-        if well_index is not None:
-            dart_well_props = map_mongo_doc_to_dart_well_props(sample)
-            set_dart_well_properties(cursor, plate_barcode, dart_well_props, well_index)
-        else:
-            raise ValueError(
-                f"Unable to determine DART well index for {sample[FIELD_ROOT_SAMPLE_ID]} in plate {plate_barcode}"
-            )
+    well_index = get_dart_well_index(str(sample.get(FIELD_COORDINATE)))
+    if well_index is not None:
+        dart_well_props = map_mongo_doc_to_dart_well_props(sample)
+        set_dart_well_properties(cursor, plate_barcode, dart_well_props, well_index)
+    else:
+        raise ValueError(
+            f"Unable to determine DART well index for {sample[FIELD_ROOT_SAMPLE_ID]} in plate {plate_barcode}"
+        )
+
+
+def add_dart_well_properties_if_positive_or_of_importance(
+    cursor: pyodbc.Cursor, sample: SampleDoc, plate_barcode: str
+) -> None:
+    """Adds well properties to DART for the specified sample if that sample is positive
+        or must_sequence or preferentially_sequence
+
+    Arguments:
+        cursor {pyodbc.Cursor} -- The cursor with which to execute queries.
+        sample {SampleDoc} -- The sample for which to add well properties.
+        plate_barcode {str} -- The barcode of the plate to which this sample belongs.
+    """
+    if is_sample_important_or_positive(sample):
```

Comment:
We have 2 function:

* add_dart_well_properties_if_important_or_positive: runs with normal crawler process in centre files (when a sample is created the first time) so we only create samples important or positive in dart.
* add_dart_well_properties: runs with the prioritisation step, it will update the status of all the wells that have changed, even when the change is to unprioritise the sample so we dont want to do the check of important or positive before running it. Only samples in priority samples will execute this, so it will only happen with samples that have changed its prioritisation,.